spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -135,6 +138,25 @@ public StreamWriter createStreamWriter(String runId, StructType dsStruct,
     return new StreamingWriter(table, io, encryptionManager, options, queryId, mode, appId, dsSchema);
   }
 
+  @Override
+  public MicroBatchReader createMicroBatchReader(Optional<StructType> schema, String checkpointLocation,
+                                                 DataSourceOptions options) {
+    if (schema.isPresent()) {
+      throw new IllegalStateException("Iceberg does not support specifying the schema at read time");
```

Comment:
This is fine for now, but we do support it on the batch read path: https://github.com/apache/incubator-iceberg/blob/master/spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java#L87-L91

Reply:
OK, let me see how it works.